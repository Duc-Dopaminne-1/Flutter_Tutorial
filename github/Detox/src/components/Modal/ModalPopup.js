import PropTypes from 'prop-types';
import React from 'react';
import {
  KeyboardAvoidingView,
  Modal,
  ModalProps,
  Platform,
  StyleSheet,
  TouchableWithoutFeedback,
  View,
  ViewPropTypes,
  ViewStyle,
} from 'react-native';

import {COLORS} from '../../assets/theme/colors';

const styles = StyleSheet.create({
  viewContainer: {
    justifyContent: 'center',
    flex: 1,
    padding: 16,
    backgroundColor: COLORS.BLACK_HALF_OPACITY,
  },
  viewOverlay: {
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    opacity: 0,
  },
});

export type Props = {
  contentContainerStyle: ViewStyle,
} & ModalProps;

const ModalPopup = ({
  children,
  onPressOutSide,
  presentationStyle = 'overFullScreen',
  transparent = true,
  animationType = 'fade',
  contentContainerStyle,
  avoidKeyboard = false,
  ...props
}: Props) => {
  return (
    <Modal
      presentationStyle={presentationStyle}
      animationType={animationType}
      transparent={transparent}
      {...props}>
      <View style={[styles.viewContainer, contentContainerStyle]}>
        <TouchableWithoutFeedback onPress={onPressOutSide}>
          <View style={styles.viewOverlay} />
        </TouchableWithoutFeedback>
        {avoidKeyboard ? (
          <KeyboardAvoidingView
            behavior={Platform.OS === 'ios' ? 'padding' : null}
            pointerEvents="box-none">
            {children}
          </KeyboardAvoidingView>
        ) : (
          children
        )}
      </View>
    </Modal>
  );
};

ModalPopup.propTypes = {
  visible: PropTypes.bool,
  onPressOutSide: PropTypes.func,
  presentationStyle: PropTypes.string, // 'fullScreen' | 'pageSheet' | 'formSheet' | 'overFullScreen',
  transparent: PropTypes.bool,
  animationType: PropTypes.string, // 'none' | 'slide' | 'fade',
  contentContainerStyle: ViewPropTypes.style,
  avoidKeyboard: PropTypes.bool,
};

ModalPopup.defaultProps = {
  visible: false,
  onPressOutSide: () => {},
  presentationStyle: 'overFullScreen',
  transparent: true,
  animationType: 'fade',
  avoidKeyboard: false,
};

export default ModalPopup;
