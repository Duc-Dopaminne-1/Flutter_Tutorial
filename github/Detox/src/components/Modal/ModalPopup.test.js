import React from 'react';
import renderer from 'react-test-renderer';

import ModalPopup from './ModalPopup';

it('renders correctly', () => {
  const tree = renderer.create(<ModalPopup />).toJSON();
  expect(tree).toMatchSnapshot();
});
