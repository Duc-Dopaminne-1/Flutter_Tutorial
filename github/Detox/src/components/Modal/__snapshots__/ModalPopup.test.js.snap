// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  presentationStyle="overFullScreen"
  transparent={true}
  visible={false}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#0005",
          "flex": 1,
          "justifyContent": "center",
          "padding": 16,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</Modal>
`;
